### PR TITLE
update association tests

### LINKED ISSUE #1255: Point TFE tests to Testnet

### LINKED ISSUE #1258: Speed up confirmations by increasing gas fee

### LINKED ISSUE #1259: - Association / Dissassociate tests - add success/confirmation to scenarios

### LINKED ISSUE #1269: Linting + Formatting in TFE webdriverio folder
- add linting and formatting to all files in the 'automation' folder in token front end 
- add formatting of code in the automation folder
- address un-needed pauses and create more implicit waiting 
- re-run tests 
- address commented out code
- address unused code / steps

### PR DESCRIPTION
**Pull request by Ditmir-Vega**
closes #1258 
closes #1259 
closes #1269 
closes #1255

### CODE REVIEW

File: automation/features/pageobjects/page.js [hunk 1]
```diff
@@ -50,12 +50,51 @@ class Page {
...
+        const state = browser.execute(function () {
+          return document.readyState;
+        });
+        return state === "complete";
+      },
+      {
+        timeout: 10000,
+        timeoutMsg: "Page did not fully load in 10 seconds",
+      }
+    );
```
Comment (dexturr) [inline automation/features/pageobjects/page.js:64]: question (blocking): why is this necessary?

File: automation/features/pageobjects/staking.page.js [hunk 2]
```diff
@@ -1,7 +1,117 @@
...
+    return browser.getByTestId("remove-stake-now-disclaimer");
+  }
+
+  associateTokens(tokenAmount, source) {
+    this.calloutContainer.waitForDisplayed({
+      timeout: 30000,
+      reverse: false,
+      timeoutMsg: "callout was not found",
+    });
+    browser.pause(1000);
```
Comment (dexturr) [inline automation/features/pageobjects/staking.page.js:70]: chore(blocking): why do we need a pause after waiting for a dom element? If not required, remove. If required, add a comment explaining why.

File: automation/features/pageobjects/staking.page.js [hunk 3]
```diff
@@ -1,7 +1,117 @@
...
+    if (this.associateMoreTokensBtn.isDisplayed()) {
+      this.associateMoreTokensBtn.click();
+    } else this.associateTokensBtn.click();
```
Comment (dexturr) [inline automation/features/pageobjects/staking.page.js:73]: chore (blocking): duplciated code

File: automation/features/pageobjects/staking.page.js [hunk 4]
```diff
@@ -1,7 +1,117 @@
...
+        break;
+      case "vesting contract":
+        this.vestingContractRadioBtn.click({ force: true });
+        break;
+      default:
+        throw new Error("Options available : [wallet , vesting contract]");
+    }
+    this.tokenAmountInputField.setValue(tokenAmount);
+    this.tokenAmountSubmitBtn.click();
+    this.confirmTransactionSpeedUp();
```
Comment (dexturr) [inline automation/features/pageobjects/staking.page.js:113]: chore (blocking): half of these exist in the page object and half don't. We should move them all to the page object

File: automation/features/pageobjects/governance.page.js [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
+    }
+    get propsalClosingDateTestId() {
+        return "governance-proposal-closingDate";
+    }
+    get proposalEnactmentDateTestId() {
+        return "governance-proposal-enactmentDate";
+    }
 }
 
 module.exports = new GovernancePage();
```
Comment (dexturr) [inline automation/features/pageobjects/governance.page.js:18]: chore (blocking): We should export the classes and create instances within all of the tests to ensure tests are deterministic and state is not leaked in between tests

File: src/routes/withdraw/eth-address-input.tsx [hunk 6]
```diff
@@ -45,12 +45,13 @@ export const EthAddressInput = ({
...
         type="button"
         onClick={() => setUseConnectedWallet(!useConnectedWallet)}
+        data-testid="enter-wallet-address-link"
         className="button-link fill"
       >
         {useConnectedWallet ? t("enterAddress") : t("useConnectedWallet")}
       </button>
 
-      {isValid ? null : <Callout intent="warn">{t("invalidAddress")}</Callout>}
+      {isValid ? null : <Callout intent="warn" datatest-id="invalid-address-callout" >{t("invalidAddress")}</Callout>}
```
Comment (dexturr) [inline src/routes/withdraw/eth-address-input.tsx:54]: chore (blocking): incorrectly specified data test id. As it is incorrect I also assume it is not used. Remove.

File: src/routes/staking/staking-form.tsx [hunk 7]
```diff
@@ -295,13 +295,14 @@ export const StakingForm = ({
                 </>
               ) : (
                 <>
-                  <p>
+                  <p >
```
Comment (dexturr) [inline src/routes/staking/staking-form.tsx:298]: chore (blocking): pointless space after p
```suggestion
                  <p>
```

File: automation/features/pageobjects/page.js (hunk 1 above)
Comment (Ditmir-Vega) [reply automation/features/pageobjects/page.js:64]: helps to alleviate some flakiness if the page is not ready and the tests aim to click on items not yet loaded

File: src/components/staking-method-radio/index.tsx [hunk 8]
```diff
@@ -18,6 +18,7 @@ export const StakingMethodRadio = ({
   return (
     <RadioGroup
       inline={true}
+      data-testid="radio-btns-container-wallet-vestingcontract"
```
Comment (dexturr) [inline src/components/staking-method-radio/index.tsx:21]: chore (blocking): unused please remove

File: src/components/locked-progress/locked-progress.tsx [hunk 9]
```diff
@@ -38,9 +38,8 @@ export const LockedProgress = ({
   return (
     <div className="tranche-item__progress">
       <div
-        className={`tranche-item__progress-bar ${
-          light ? "tranche-item__progress-bar--light" : ""
-        }`}
+        className={`tranche-item__progress-bar ${light ? "tranche-item__progress-bar--light" : ""
```
Comment (dexturr) [inline src/components/locked-progress/locked-progress.tsx:41]: chore (blocking): bunch of pointless whitespace changes. Can you ensure that prettier is installed as an extension in your IDE please.

Also can you revert these changes and this is the incorrect formatting.

File: package.json [hunk 10]
```diff
@@ -76,6 +77,8 @@
     "rules": {
       "simple-import-sort/imports": "warn",
       "no-unused-vars": "off",
+      "await-expect": "off",
+      "@wdio/no-undef": "off",
```
Comment (dexturr) [inline package.json:81]: chore (blocking): should not be defined here.
```suggestion
```

File: package.json [hunk 11]
```diff
@@ -62,6 +62,7 @@
     "build": "craco build",
     "test": "REACT_APP_CHAIN=0x1 craco test",
     "lint:es": "eslint . --ext .ts,.tsx,.js --max-warnings 0",
+    "lint:es:fix": "eslint --fix . --ext .ts,.tsx,.js --max-warnings 0",
```
Comment (dexturr) [inline package.json:65]: chore (blocking): remove this, it's no needed. When using yarn arguments are passed down so `yarn lint:es --fix` is the same as this 
```suggestion
```

File: automation/wdio.conf.js [hunk 12]
```diff
@@ -88,16 +88,19 @@ exports.config = {
...
       browser.addCommand(key, commands[key]);
     });
     browser.switchWindow("Vega");
   },
 
-  afterStep: function (/* step, scenario, result, context */) {
+  afterStep: function () {
+    // if (!result.passed){
     browser.takeScreenshot();
+    // }
```
Comment (dexturr) [inline automation/wdio.conf.js:103]:
```suggestion
```

File: automation/wdio.conf.js [hunk 13]
```diff
@@ -88,16 +88,19 @@ exports.config = {
     ignoreUndefinedDefinitions: false,
   },
 
-  before: function (/* capabilities, specs */) {
+  before: function () {
     // Add commands to WebdriverIO
+    /* eslint-disable no-unused-expressions */
```
Comment (dexturr) [inline automation/wdio.conf.js:93]: chore (blocking): this eslint disable is doing nothing
```suggestion
```

File: automation/features/rewards.feature [hunk 14]
```diff
@@ -1,17 +1,20 @@
...
-    Then I can see the rewards epoch timer displayed
-    And I can see the connected vega wallet key
-    And I disconnect Vega Wallet
+    Then the connect to vega wallet is shown
+    And the rewards epoch countdown is not displayed
+
+  Scenario: Rewards are shown and rewards countdown is shown when vega wallet is connect
+    When I connect to the vega wallet
+    Then the epoch countdown timer is counting down
+    # And the correct vega key is shown for rewards
```
Comment (dexturr) [inline automation/features/rewards.feature:14]: chore (blocking): fix or remove

File: automation/features/pageobjects/staking.page.js (hunk 3 above)
Comment (dexturr) [inline automation/features/pageobjects/staking.page.js:73]: chore (blocking): this formatting on if statements through the PR is not what is normally used. Please change to always use curly braces. Installing prettier for your IDE should auto fix this for you.

File: automation/features/pageobjects/staking.page.js (hunk 2 above)
Comment (Ditmir-Vega) [reply automation/features/pageobjects/staking.page.js:70]: The callout is the container that holds the 'Associate' 'disassociate' buttons in this case . some times the container will load and the button will not load straight away with it. 
we also can't assert to wait for the other buttons to appear because  we might have either associate or associate more button displayed 

Only option i can see is maybe waiting for the 'disassociate button to display as that is always there

File: automation/features/step-definitions/staking.steps.js [hunk 15]
```diff
@@ -1,45 +1,72 @@
...
-  const nodeList = $(".node-list").$$('[data-testid="node-list-item"]');
-  expect(nodeList.length).toBeGreaterThan(0);
-});
+When(
+  /^I associate "([^"]*)?" tokens from "([^"]*)?"$/,
+  (tokenAmount, sourceOfFunds) => {
+    // $(".tranche-item__progress").waitForDisplayed({
+    //   timeout: 5000,
+    //   timeoutMsg: "wallet card did not fully render in 5 seconds",
+    // });
```
Comment (dexturr) [inline automation/features/step-definitions/staking.steps.js:18]:
```suggestion
```

File: automation/features/step-definitions/staking.steps.js [hunk 16]
```diff
@@ -1,45 +1,72 @@
 const { When, Then } = require("@wdio/cucumber-framework");
+const stakingPage = require("../pageobjects/staking.page");
 let currentAmountOfTokensInWallet = "";
+let validatorName = "";
+/* eslint-disable  no-unused-vars */
+let tokensAssociatedInVegaWallet = "";
```
Comment (dexturr) [inline automation/features/step-definitions/staking.steps.js:6]: chore (blocking): Remove unused vatrs don't just disable the linting rules for them. all of thse in this file

File: automation/features/step-definitions/staking.steps.js [hunk 17]
```diff
@@ -1,45 +1,72 @@
...
-
-When(/^I can see the pending transactions button is shown$/, async () => {
-  browser.getByTestId("pending-transactions-btn").waitForDisplayed({
+When(
+  /^I disassociate "([^"]*)?" tokens from "([^"]*)?"$/,
+  (tokenAmount, sourceOfFunds) => {
+    // $(".tranche-item__progress").waitForDisplayed({
+    //   timeout: 5000,
+    //   timeoutMsg: "wallet card did not fully render in 5 seconds",
+    // });
```
Comment (dexturr) [inline automation/features/step-definitions/staking.steps.js:42]:
```suggestion
```

File: automation/features/step-definitions/staking.steps.js [hunk 18]
```diff
@@ -48,23 +75,321 @@ When(/^I click on use maximum button$/, () => {
...
+    .getText();
   browser.getByTestId("token-amount-use-maximum").click();
 });
 
 When(
   /^I can see the maximum amount of tokens in my wallet are in the token input box$/,
   () => {
     const noZeroes = currentAmountOfTokensInWallet.toString();
-    const noZeroesFloat = parseFloat(noZeroes);
+    const noZeroesFloat = parseFloat(noZeroes.replace(",", ""));
```
Comment (dexturr) [inline automation/features/step-definitions/staking.steps.js:92]: chore (blocking): badly named variable, suggest token balance

File: automation/features/pageobjects/staking.page.js (hunk 3 above)
Comment (Ditmir-Vega) [reply automation/features/pageobjects/staking.page.js:73]: where is the other duplication ?

File: automation/features/step-definitions/staking.steps.js [hunk 19]
```diff
@@ -48,23 +75,321 @@ When(/^I click on use maximum button$/, () => {
...
+});
+
+When(/^I click on a validator$/, () => {
+  $('.node-list__item-name"]').waitForDisplayed({
+    timeout: 30000,
+    reverse: false,
+    timeoutMsg: "Validator list not displayed",
+  });
+  validatorName = $$(".node-list__item-name")[0].getText();
+  $$(".node-list__item-name")[0].click();
```
Comment (dexturr) [inline automation/features/step-definitions/staking.steps.js:123]: chore (blocking): half of these steps are in the stakingPage and half aren't. We should be consistent.

File: automation/features/step-definitions/staking.steps.js [hunk 20]
```diff
@@ -48,23 +75,321 @@ When(/^I click on use maximum button$/, () => {
...
     timeoutMsg: "callout was not found",
   });
-  if (browser.getByTestId("associate-more-tokens-btn").isDisplayed()) {
-    browser.getByTestId("associate-more-tokens-btn").click();
-  } else {
-    browser.getByTestId("associate-tokens-btn").click();
-  }
+  if (stakingPage.associateMoreTokensBtn.isDisplayed()) {
+    stakingPage.associateMoreTokensBtn.click();
+  } else stakingPage.associateTokensBtn.click();
```
Comment (dexturr) [inline automation/features/step-definitions/staking.steps.js:80]: chore (blocking): Duplicated code. Probably would move this to a facade step in the staking page.

File: automation/features/step-definitions/staking.steps.js [hunk 21]
```diff
@@ -48,23 +75,321 @@ When(/^I click on use maximum button$/, () => {
...
+  const nodeList = $(".node-list").$$('[data-testid="node-list-item"]');
+  expect(nodeList.length).toBeGreaterThan(0);
+});
+
+Then(/^the epoch countdown timer is counting down$/, () => {
+  const currentCountdownTimerText = browser
+    .getByTestId("current-epoch-ends-in")
+    .getText();
+  /* eslint-disable wdio/no-pause */
+  browser.pause(2100); // let some time pass
```
Comment (dexturr) [inline automation/features/step-definitions/staking.steps.js:110]:
```suggestion
  browser.pause(2100);
```

Comment (dexturr) [reply automation/features/step-definitions/staking.steps.js:110]: Why is this pause needed? Can we not wait for something in the dom?

File: automation/features/pageobjects/governance.page.js [hunk 22]
```diff
@@ -1,18 +1,18 @@
...
-    return "governance-proposal-table";
-  }
-  get propsalClosingDateTestId() {
-    return "governance-proposal-closingDate";
-  }
-  get proposalEnactmentDateTestId() {
-    return "governance-proposal-enactmentDate";
-  }
+    get proposalStatesTestId() {
+        return "governance-proposal-state";
```
Comment (mattrussell36) [inline automation/features/pageobjects/governance.page.js:5]: Indentation off here

File: automation/features/step-definitions/shared-steps.steps.js [hunk 23]
```diff
@@ -23,17 +23,15 @@ When(/^I connect Vega wallet$/, () => {
...
 
-Then(/^I am taken to the "([^"]*)?" page$/, async (page) => {
+Then(/^I am taken to the "([^"]*)?" page$/, (page) => {
   if (page === "home") {
-    await expect(browser).toHaveUrl(browser.options.baseUrl + "/");
-  } else {
-    await expect(browser).toHaveUrlContaining("/" + page);
-  }
+    expect(browser).toHaveUrl(browser.options.baseUrl + "/");
+  } else expect(browser).toHaveUrlContaining("/" + page);
```
Comment (mattrussell36) [inline automation/features/step-definitions/shared-steps.steps.js:29]: Curly braces for else block

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: automation/wdio.conf.js (hunk 13 above)
Comment (Ditmir-Vega) [reply automation/wdio.conf.js:93]: done

File: automation/wdio.conf.js (hunk 12 above)
Comment (Ditmir-Vega) [reply automation/wdio.conf.js:103]: done

File: automation/wdio.conf.js [hunk 24]
```diff
@@ -88,16 +88,19 @@ exports.config = {
...
     Object.keys(commands).forEach((key) => {
       browser.addCommand(key, commands[key]);
     });
     browser.switchWindow("Vega");
   },
 
-  afterStep: function (/* step, scenario, result, context */) {
+  afterStep: function () {
+    // if (!result.passed){
```
[the remark this answers is not in the document]
Comment (Ditmir-Vega) [reply automation/wdio.conf.js:101]: removed . But normally if we have alot of tests we would switch this back on so it only takes screenshots when the test result is 'failed'

File: package.json (hunk 11 above)
Comment (Ditmir-Vega) [reply package.json:65]: done

File: src/components/staking-method-radio/index.tsx (hunk 8 above)
Comment (Ditmir-Vega) [reply src/components/staking-method-radio/index.tsx:21]: done

File: automation/features/tranches.feature [hunk 25]
```diff
@@ -1,13 +1,14 @@
 Feature: Tranches page
 
-  @ignore
+@ignore
   Scenario: Go to Tranches page
     Given I connect my ethereum wallet
     And I navigate to "tranches" page
     Then I am taken to the "tranches" page
     # Then I can see the header title is "Vesting tranches"
```
[the remark this answers is not in the document]
Comment (Ditmir-Vega) [reply automation/features/tranches.feature:8]: added @todo tag instead and uncommented the step

File: automation/features/rewards.feature (hunk 14 above)
Comment (Ditmir-Vega) [reply automation/features/rewards.feature:14]: removed - done

File: automation/features/step-definitions/staking.steps.js (hunk 15 above)
Comment (Ditmir-Vega) [reply automation/features/step-definitions/staking.steps.js:18]: done - removed

File: automation/features/step-definitions/staking.steps.js (hunk 16 above)
Comment (Ditmir-Vega) [reply automation/features/step-definitions/staking.steps.js:6]: returns unused but we actually assign values to them so not sure why it still returns 'unused vars'

File: automation/features/step-definitions/staking.steps.js (hunk 17 above)
Comment (Ditmir-Vega) [reply automation/features/step-definitions/staking.steps.js:42]: done -removed

File: automation/features/step-definitions/staking.steps.js (hunk 18 above)
Comment (Ditmir-Vega) [reply automation/features/step-definitions/staking.steps.js:92]: done renamed to tokenBalance and tokenBalanceFloat

File: src/components/locked-progress/locked-progress.tsx (hunk 9 above)
Comment (Ditmir-Vega) [reply src/components/locked-progress/locked-progress.tsx]: done

File: automation/features/pageobjects/staking.page.js (hunk 3 above)
Comment (Ditmir-Vega) [reply automation/features/pageobjects/staking.page.js]: done and fixed in some other areas too to add curly brace

File: automation/features/pageobjects/governance.page.js (hunk 22 above)
Comment (Ditmir-Vega) [reply automation/features/pageobjects/governance.page.js]: done

File: automation/features/step-definitions/shared-steps.steps.js (hunk 23 above)
Comment (Ditmir-Vega) [reply automation/features/step-definitions/shared-steps.steps.js]: done

File: automation/features/pageobjects/governance.page.js (hunk 5 above)
Comment (Ditmir-Vega) [reply automation/features/pageobjects/governance.page.js]: we can have a ticket for this to look into it but it's not part of this work

File: automation/package.json [hunk 26]
```diff
@@ -1,26 +1,26 @@
...
+    "@wdio/sync": "^7.16.13",
+    "chromedriver": "^99.0.0",
+    "wdio-chromedriver-service": "^7.2.6"
+  },
+  "scripts": {
+    "wdio": "wdio run wdio.conf.js",
+    "wdio:local": "wdio run wdio.conf.local.js",
+    "wdio:local:test": "wdio run wdio.conf.local.js --cucumberOpts.tagExpression='@test'",
+    "yarn:start": "cd .. && yarn start",
+    "yarn:install:deps": "cd .. && yarn"
```
Comment (dexturr) [inline automation/package.json:20]: follow up (non-blocking): can you please ticket the following:

* This script is pointless and vanity they should be removed and the workflow should just inline the commands
* This package is using NPM and should be moved to using yarn

File: automation/features/pageobjects/governance.page.js (hunk 5 above)
Comment (dexturr) [reply automation/features/pageobjects/governance.page.js]: Which ticket?

File: automation/features/step-definitions/staking.steps.js [hunk 27]
```diff
@@ -48,23 +68,323 @@ When(/^I click on use maximum button$/, () => {
...
     timeoutMsg: "callout was not found",
   });
-  if (browser.getByTestId("associate-more-tokens-btn").isDisplayed()) {
-    browser.getByTestId("associate-more-tokens-btn").click();
+  if (stakingPage.associateMoreTokensBtn.isDisplayed()) {
+    stakingPage.associateMoreTokensBtn.click();
   } else {
-    browser.getByTestId("associate-tokens-btn").click();
+    stakingPage.associateTokensBtn.click();
   }
```
Comment (dexturr) [inline automation/features/step-definitions/staking.steps.js:75]: In staking page you have this:

```
    if (this.associateMoreTokensBtn.isDisplayed()) {
      this.associateMoreTokensBtn.click();
    } else {
      this.associateTokensBtn.click();
    }
```
Let's create a function in the staking page called "clickOnAssociation" or similar that does this and just call it here. Removes the duplicated logic :D

File: automation/features/step-definitions/staking.steps.js [hunk 28]
```diff
@@ -1,45 +1,65 @@
...
-  browser.pause(2000);
-});
-
-When(/^I can see the pending transactions button is shown$/, async () => {
-  browser.getByTestId("pending-transactions-btn").waitForDisplayed({
+When(
+  /^I disassociate "([^"]*)?" tokens from "([^"]*)?"$/,
+  (tokenAmount, sourceOfFunds) => {
+    /* eslint-disable  wdio/no-pause */
+    browser.pause(500);
```
Comment (dexturr) [inline automation/features/step-definitions/staking.steps.js:37]: Either remove or add a comment why this is needed.

File: automation/features/step-definitions/staking.steps.js [hunk 29]
```diff
@@ -48,23 +68,323 @@ When(/^I click on use maximum button$/, () => {
...
+  const nodeList = $(".node-list").$$('[data-testid="node-list-item"]');
+  expect(nodeList.length).toBeGreaterThan(0);
+});
+
+Then(/^the epoch countdown timer is counting down$/, () => {
+  const currentCountdownTimerText = browser
+    .getByTestId("current-epoch-ends-in")
+    .getText();
+  /* eslint-disable wdio/no-pause */
+  browser.pause(2100); // let some time pass
```
Comment (dexturr) [inline automation/features/step-definitions/staking.steps.js:105]: Let's add a comment to that effect then

```suggestion
  browser.pause(2100); // let some time pass so the epoch timer advances
```

File: automation/features/pageobjects/page.js [hunk 30]
```diff
@@ -42,20 +42,59 @@ class Page {
...
   }
 
   connectToEthWallet() {
     this.connectEthWallet.click();
     this.connectMetaMask.click();
   }
+
+  confirmTransactionSpeedUp() {
+    browser.waitUntil(() => browser.getWindowHandles().length > 1);
+    browser.switchWindow("MetaMask Notification");
```
Comment (dexturr) [inline automation/features/pageobjects/page.js:74]: I'd like to make a class for meta mask interactions. I understand this is on the base page class, which conceptualy is elements always on the page under test. So would be cool to have:


```
class MM {
   setup () {/* Do things required to import network ect. */}
   confirm () {/* Do things required to import network ect. */}
   viewAssetList () {/* Do things required to import network ect. */}
}
```

view asset list will be needed for some stuff relating to adding tokens to wallet later. 

Can do this as a follow up if you don't want to do it in this PR, else create an issue and we can do soon.

**Review by dexturr — CHANGES_REQUESTED** [review]
Looks like a test is failing, can we make this less flakey/fix ect 👍 

Screenshot 2022-04-19 at 10 50 08

File: automation/features/pageobjects/page.js (hunk 30 above)
Comment (Ditmir-Vega) [reply automation/features/pageobjects/page.js:74]: we can do this as a follow up would be better thanks

Comment (Ditmir-Vega) [reply automation/features/pageobjects/page.js:74]: https://github.com/vegaprotocol/token-frontend/issues/1281

File: automation/features/pageobjects/governance.page.js (hunk 5 above)
Comment (Ditmir-Vega) [reply automation/features/pageobjects/governance.page.js]: https://github.com/vegaprotocol/token-frontend/issues/1282

**Comment by Ditmir-Vega** [comment]
Screenshot 7yhuyhsu8xuyh
Staking test on testnet

**Comment by Ditmir-Vega** [comment]
on the latest report : https://vegaprotocol.github.io/token-frontend/206/ known tests with staking are timing out waiting for confrirmations

File: automation/features/step-definitions/staking.steps.js (hunk 21 above)
Comment (vega-paul) [reply automation/features/step-definitions/staking.steps.js:105]: Ditmir will add support for epoch wait check using API which provides determinstic wait time

File: automation/features/step-definitions/staking.steps.js (hunk 28 above)
Comment (Ditmir-Vega) [reply automation/features/step-definitions/staking.steps.js]: this pause has been removed

File: automation/features/step-definitions/staking.steps.js [hunk 31]
```diff
@@ -1,70 +1,349 @@
...
-  // eslint-disable-next-line wdio/no-pause
-  browser.pause(2000);
-});
-
-When(/^I can see the pending transactions button is shown$/, async () => {
-  browser.getByTestId("pending-transactions-btn").waitForDisplayed({
+When(
+  /^I disassociate "([^"]*)?" tokens from "([^"]*)?"$/,
+  (tokenAmount, sourceOfFunds) => {
+    tokensAssociatedInVegaWallet = tokensAssociatedInVegaWalletText.replace(
```
Comment (dexturr) [inline automation/features/step-definitions/staking.steps.js:17]: chore (blocking): Linting errors, please fix. There are 2 others in this PR

Comment (dexturr) [inline automation/features/step-definitions/staking.steps.js:17]: chore (blocking): Linting errors, please fix. There are 2 others in this PR

File: .github/workflows/test.yml [hunk 32]
```diff
@@ -6,119 +6,119 @@ on:
...
+      #   run: curl -L https://github.com/vegaprotocol/vegawallet/releases/download/v0.13.1/vegawallet-linux-amd64.zip -O
+      #   # run: curl -L https://github.com/vegaprotocol/vegawallet/releases/download/v${{ matrix.vegawallet-version }}/vegawallet-linux-amd64.zip -O
+      # - name: ls current
+      #   run: ls
+      # - name: ls previous
+      #   run: ls ..
+      # - name: Unzip wallet
+      #   run: unzip vegawallet-linux-amd64.zip
+      - name: Download and Unzip latest wallet
+        run: curl -s https://api.github.com/repos/vegaprotocol/vegawallet/releases/latest | awk -F\" '/browser_download_url.*linux-amd64.zip/{print $(NF-1)}' | xargs wget && unzip vegawallet-linux-amd64.zip && rm vegawallet-linux-amd64.zip
```
Comment (dexturr) [inline .github/workflows/test.yml:36]: chore (blocking): pipeline should be deterministic using latest means that it is not. Let's use the same logic as in the frontend monorepo please and fix to a matrix defined version of the wallet:

https://github.com/vegaprotocol/frontend-monorepo/blob/master/.github/workflows/cypress.yml#L16-L17

and consuming that in this curl:
https://github.com/vegaprotocol/frontend-monorepo/blob/master/.github/workflows/cypress.yml#L38-L39

File: .github/workflows/test.yml [hunk 33]
```diff
@@ -6,119 +6,119 @@ on:
...
+        run: npm install
+      # - name: Download wallet
+      #   run: curl -L https://github.com/vegaprotocol/vegawallet/releases/download/v0.13.1/vegawallet-linux-amd64.zip -O
+      #   # run: curl -L https://github.com/vegaprotocol/vegawallet/releases/download/v${{ matrix.vegawallet-version }}/vegawallet-linux-amd64.zip -O
+      # - name: ls current
+      #   run: ls
+      # - name: ls previous
+      #   run: ls ..
+      # - name: Unzip wallet
+      #   run: unzip vegawallet-linux-amd64.zip
```
Comment (dexturr) [inline .github/workflows/test.yml:34]: chore (blocking): delete

File: .github/workflows/test.yml [hunk 34]
```diff
@@ -6,119 +6,119 @@ on:
...
-        #       context: 'Test report'
-        #       state: 'success'
-        #       sha: ${{ github.event.pull_request.head.sha }}
-        #       target_url: https://vegaprotocol.github.io/token-frontend/${{ github.run_number }}
+      - name: "BrowserStack Local Tunnel Setup" # Invokes the setup-local action
+        uses: browserstack/github-actions/setup-local@master
+        with:
+          local-testing: start
+        # with:
+        #   wait-on: http://localhost:3000
```
Comment (dexturr) [inline .github/workflows/test.yml:65]: Delete

File: .github/workflows/test.yml [hunk 35]
```diff
@@ -6,119 +6,119 @@ on:
...
+          BROWSERSTACK_KEY: ${{ secrets.BROWSERSTACK_KEY }}
+      # - name: 'BrowserStackLocal Stop'  # Terminating the BrowserStackLocal tunnel connection
+      #   uses: browserstack/github-actions/setup-local@master
+      #   with:
+      #     local-testing: stop
+      #   - uses: actions/upload-artifact@v1
+      #     if: always()
+      #     with:
+      #       name: logs
+      #       path: logs
```
Comment (dexturr) [inline .github/workflows/test.yml:81]: Delete

File: .github/workflows/test.yml [hunk 36]
```diff
@@ -6,119 +6,119 @@ on:
...
+        if: always()
+        continue-on-error: true
+        with:
+          ref: gh-pages
+          path: gh-pages
+      # Generate report
+      - name: Allure Report action from marketplace
+        uses: simple-elf/allure-report-action@master
+        if: always()
+        #id: allure-report
```
Comment (dexturr) [inline .github/workflows/test.yml:93]:
```suggestion
        id: allure-report
```

File: .github/workflows/test.yml [hunk 37]
```diff
@@ -6,119 +6,119 @@ on:
...
+        with:
+          allure_results: automation/allure-results
+          #gh_pages: gh-pages
+          #allure_report: allure-report
```
Comment (dexturr) [inline .github/workflows/test.yml:97]: question (blocking): Do we need this back in?

File: .github/workflows/test.yml [hunk 38]
```diff
@@ -6,119 +6,119 @@ on:
...
+      #     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      # - name: Post the link to the report in PR comment
+      #   if: always()
+      #   uses: Sibz/github-status-action@v1
+      #   with:
+      #       authToken: ${{secrets.GITHUB_TOKEN}}
+      #       context: 'Test report'
+      #       state: 'success'
+      #       sha: ${{ github.event.pull_request.head.sha }}
+      #       target_url: https://vegaprotocol.github.io/token-frontend/${{ github.run_number }}
```
Comment (dexturr) [inline .github/workflows/test.yml:124]: question (blocking): Would be good if we can have this back in else removed

File: automation/features/step-definitions/home.steps.js [hunk 39]
```diff
@@ -50,10 +50,14 @@ Then(/^staked token field is showing as "([^"]*)?"$/, (amount) => {
   expect(stakedToken.getText()).toBe(amount);
 });
 
-Then(/^the vega wallet link is correct$/, async () => {
-  await expect(HomePage.vegaWalletLink).toHaveAttribute(
+Then(/^the vega wallet link is correct$/, () => {
+  HomePage.vegaWalletLink.waitForDisplayed({
```
Comment (dexturr) [inline automation/features/step-definitions/home.steps.js:54]: chore (blocking): HomePage here is an instance and not a class so it should be camelCase not PascalCase

File: automation/features/step-definitions/staking.steps.js [hunk 40]
```diff
@@ -1,70 +1,349 @@
...
+        timeout: 60000,
+        timeoutMsg: "expected balance to be different",
+      }
+    )
+  ).toBeTruthy();
+});
+
+When(
+  /^the disassociation of "([^"]*)?" has been successful$/,
+  (tokenAmount) => {
```
Comment (dexturr) [inline automation/features/step-definitions/staking.steps.js:273]: question (blocking): don't we need to do some maths here with the tokenAmount and check balances? Same for linting error above?

Comment (dexturr) [reply automation/features/step-definitions/staking.steps.js:273]: Or a TODO comment to add this when we can do this determinisitically?

File: automation/package.json (hunk 26 above)
Comment (dexturr) [reply automation/package.json:20]: Yes but:
1. This project is not using yarn so exposing it as a dependency in this package JSON does not make sense
2. This is a vanity script and could just be inline into the workflow file without the need for these
3. These are not doing anything complex (2 commands) so I would expect these to be run in line in the workflow file. If they were calling more things then that would be fine

So can we add a ticket to tidy this up please, it will only take 5 mins to do :+1:

**Review by dexturr — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 38 above)
Comment (Ditmir-Vega) [reply .github/workflows/test.yml:124]: REMOVED

File: .github/workflows/test.yml (hunk 37 above)
Comment (Ditmir-Vega) [reply .github/workflows/test.yml:97]: No

File: automation/features/step-definitions/staking.steps.js (hunk 27 above)
Comment (Ditmir-Vega) [reply automation/features/step-definitions/staking.steps.js:40]: raised as issue

File: automation/features/staking.feature [hunk 41]
```diff
@@ -1,68 +1,100 @@
...
+    Given I navigate to "staking" page
 
-    Scenario: Associate tokens through wallet
-      Given I navigate to "staking" page
-      When I connect to the vega wallet
-      And I associate some tokens from wallet
-      Then I can see the pending transactions button is shown
+  Scenario: Go to staking and connect ethereum wallet
+    Given I connect my ethereum wallet
+    Then I can see my ethereum key 0x9804C6E98dA75e3271ccE3aA56728FD8e9376155 is shown
```
Comment (jtsang586) [inline automation/features/staking.feature:8]: Non-blocking: this is another one that should use a magic string

File: automation/features/pageobjects/staking.page.js [hunk 42]
```diff
@@ -1,7 +1,171 @@
...
+    return $(
+      '//*[@id="root"]/div/div[2]/aside/section[1]/div/div[2]/div[7]/div[3]/span[1]'
+    ).getText();
+  }
+
+  getByLabelText(labelText) {
+    return $(`label=${labelText}`);
+  }
+
+  getBypTag(pText) {
```
Comment (jtsang586) [inline automation/features/pageobjects/staking.page.js:112]: Typo?

Comment (Ditmir-Vega) [reply automation/features/pageobjects/staking.page.js:112]: no , this is correct get by <p> tag basically 
There are alot of places where there is no id or test id or classes just a <p> tag

**Review by jtsang586 — APPROVED** [review]

**Review by mattrussell36 — APPROVED** [review]
